language-support/hs/bindings/src/DA/Ledger/LedgerService.hs

```diff
@@ -4,52 +4,52 @@
 -- Abstraction for LedgerService, which can be composed monadically.
 module DA.Ledger.LedgerService (
     LedgerService, runLedgerService, makeLedgerService, TimeoutSeconds,
-    Jwt, setToken,
+    setToken,
     askTimeout,
     ) where
 
 import Control.Monad.Fail (MonadFail)
 import Control.Monad.IO.Class (MonadIO)
 import Control.Monad.Reader (MonadReader,local,asks)
 import Control.Monad.Trans.Reader (ReaderT(..))
-import DA.Ledger.Jwt (Jwt)
 import DA.Ledger.Retry (ledgerRetry)
 import Network.GRPC.HighLevel.Client(TimeoutSeconds)
 import Network.GRPC.HighLevel.Generated(ClientConfig,MetadataMap(..))
 import UnliftIO(MonadUnliftIO)
-import qualified DA.Ledger.Jwt as Jwt (toBearerString)
 import qualified Data.ByteString.UTF8 as BSU8
 import qualified Data.Map as Map
 import qualified Data.SortedList as SortedList
 
 data Context = Context
   { ts :: TimeoutSeconds
   , cc :: ClientConfig
-  , jwtMaybe :: Maybe Jwt
+  , tokMaybe :: Maybe Tok
   }
 
+type Tok = String
```

Comment:
Please avoid type synonyms for this kind of stuff. Either make it a newtype to make sure that you actually get some safety or use `String`. Making it a type synonym just makes the API harder to use since I still have to look at the definition to figure out what I can pass while not providing any additional safety or readability, e.g., when I see `setToken "foobar" ledgerService` I can at best guess what the string is where is if I see `setToken (Token "foobar") ledgerService` that’s much clearer.